src/benchmark_register.cc

```diff
@@ -293,6 +305,16 @@ Benchmark* Benchmark::Ranges(const std::vector<std::pair<int, int>>& ranges) {
   return this;
 }
 
+Benchmark* Benchmark::ArgName(const std::string& name) {
+  arg_names_ = {name};
+  return this;
+}
+
+Benchmark* Benchmark::ArgNames(const std::vector<std::string>& names) {
+  arg_names_ = names;
```

Comment:
I've added the constraint for the size. I've also modified ArgsCnt, so that it could be possible to call `Args` and `ArgNames` in any order. If you see a simpler/more elegant way to do this, I'm all ears.